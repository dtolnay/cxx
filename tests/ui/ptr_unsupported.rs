#[cxx::bridge]
mod ffi {
    unsafe extern "C++" {
        type C;

        fn get_ptr_to_reference() -> *mut &C;
        fn get_uniqueptr_to_ptr() -> UniquePtr<*mut C>;
    }
}

fn main() {}
