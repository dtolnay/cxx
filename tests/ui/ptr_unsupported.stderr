error: C++ does not allow pointer to reference as a type
 --> tests/ui/ptr_unsupported.rs:6:38
  |
6 |         fn get_ptr_to_reference() -> *mut &C;
  |                                      ^^^^^^^

error: unsupported unique_ptr target type
 --> tests/ui/ptr_unsupported.rs:7:38
  |
7 |         fn get_uniqueptr_to_ptr() -> UniquePtr<*mut C>;
  |                                      ^^^^^^^^^^^^^^^^^
